File: Number Theory - Prime Generate.cpp
#include <bits/stdc++.h>
using namespace std;
vector<int> generate_primes(int n) {
    vector<bool> is_prime(n+1);
    vector<int> primes;
    fill(is_prime.begin(), is_prime.end(), true);
    is_prime[0] = is_prime[1] = false;
    for(int i = 2; i <= n; i++) {
        if(is_prime[i]) {
            primes.push_back(i);
            for(int j = 2 * i; j <= n; j += i) {
                is_prime[j] = false;
            }
        }
    }
    return primes;
}

File: Number Theory - Prime Generate.py
def generate_primes(n):
    is_prime = [False if i <= 1 else True for i in range(n + 1)]
    primes = []
    for i in range(2, n + 1):
        if is_prime[i]:
            primes.append(i)
            for j in range(2 * i, n + 1, i):
                is_prime[j] = False
    return primes
